### ISSUE TITLE
Feature Request: Original Cpm in targeting

### ISSUE THREAD

**Issue by Guilemon** [prebid/prebid-server#1526]
Currently, it is difficult to track original bid cpm if currency conversion or bid adjustments are enabled for an adunit in prebid server. Also, in prebid universal creative for AMP, the ${AUCTION_PRICE} macro will be replaced by hb_pb value as per [this pull request](https://github.com/prebid/prebid-universal-creative/pull/62), which in current case might be currency converted or bid adjusted cpm instead of original bid cpm that was made by the bidder.

**Comment by SyntaxNode** [comment]
Is this request for debugging purposes?

**Comment by Guilemon** [comment]
Yes, the original cpm is needed for debugging purposes

**Comment by SyntaxNode** [comment]
@Guilemon Thank you for clarifying. This sounds like a feature which may only be activated when the debug or test flags are set.

Where do you propose the original cpm is stored in the request?

**Comment by Guilemon** [comment]
I was hoping it to be added in targeting itself, so that side by side comparison can be made with modified hb_pb values and original bid cpm

**Comment by bretg** [comment]
Discussed in Prebid Server committee -- we're not in favor of adding targeting in general. Many pubs already have trouble with too many ad server variables.

So, if ad server targeting for AMP is the key part of this @Guilemon, we think this can be solved with https://github.com/prebid/prebid-server/issues/1167 -- this allows the additional targeting to be applied to only those that want it. Problem is that this is a pretty big change and there's confirmed timeline for it.

If, on the other hand, it would be useful to have the original CPM in the OpenRTB response, we're willing to add seatbid.bid.ext.origbidcpm and origbidcur

**Comment by bretg** [comment]
Confirmed that the configurable ad server targeting is available in PBS-Java as of 1.41. 

At a high level, you would add something like this to the AMP Stored Requests to get custom targeting:
```
   ext.prebid.adservertargeting:  [{
               "key": "origcpm_{{BIDDER}}", // {{BIDDER}} is a macro to be resolved
               "source": "bidresponse", 
               "value": "seatbid.bid.price"
            }
   }]
```
However, I suspect this won't quite meet your use case as seatbid.bid.price has probably already been overwritten by the time targeting is generated. So we would need to build the seatbid.bid.ext.origbidcpm/origbidcur feature before you could really do what's needed:
```
   ext.prebid.adservertargeting:  [{
               "key": "origcpm_{{BIDDER}}", // {{BIDDER}} is a macro to be resolved
               "source": "bidresponse", 
               "value": "seatbid.bid.ext.origbidcpm"
            }
   }]
```

However, there's no committed timeline for this feature in PBS-Go.

**Comment by bretg** [comment]
@Guilemon - we're working on building`seatbid.bid.ext.origbidcpm` in PBS-Java. The question came up about whether this is desired as a feature that's on all the time or only in debug/test mode.

I'm guessing that you'd want this all the time as part of a regular check up? This feature adds a couple new fields to every seatbid.bid.ext. Questions:

1) Are these fields desired even if there was no bid adjustment made?
2) Is there a need to control the addition of these response variables per account?

**Comment by Guilemon** [comment]
@bretg  Apart from bid adjustment, original bid cpm is also required when currency conversion is enabled. Other times, it doesn't seem it will be required. 
I'm not sure if there will be any case where a control will be required for adding these variables on per account basis

**Comment by bretg** [comment]
@SyntaxNode - any objection to just adding `seatbid.bid.ext.origbidcpm` all the time? Seems cheap enough.

**Comment by bretg** [comment]
Approved by Prebid Server Committee

**Comment by bretg** [comment]
this was done in PBS-Java a while back.

**Comment by SyntaxNode** [comment]
Implemented in PBS-Go 0.208.0.
